apps/renderscript/test_blur_copy.cpp

```diff
@@ -2,13 +2,104 @@
 
 using namespace Halide;
 
-void blur(std::string suffix, ImageParam input8, const int channels) {
+void blur_float(std::string suffix, ImageParam input8, const int channels) {
     Var x, y, c;
     Func input;
     input(x, y, c) = input8(clamp(x, input8.left(), input8.right()),
                             clamp(y, input8.top(), input8.bottom()), c);
 
     Func blur_x("blur_x");
+    blur_x(x, y, c) = (input(x, y, c) +
+        input(x + 1, y, c) +
+        input(x + 2, y, c)) / 3;
+
+    Func result("result");
+    result(x, y, c) = (blur_x(x, y, c) +
+        blur_x(x, y + 1, c) +
+        blur_x(x, y + 2, c)) / 3;
+
+    // Unset default constraints so that specialization works.
+    result.output_buffer().set_stride(0, Expr());
+
+    result.bound(c, 0, channels);
+
+    Expr interleaved =
+        (result.output_buffer().stride(0) == channels &&
+         result.output_buffer().stride(2) == 1);
+    Expr planar = result.output_buffer().stride(0) == 1;
+
+    if (suffix == "_rs") {
+        result.shader(x, y, c, DeviceAPI::Renderscript);
+        result.specialize(interleaved).vectorize(c);
+        // non-specialized version is planar
+    } else {
+        Var yi;
+        result
+            .reorder(c, x, y)
+            .unroll(c)
+            .split(y, y, yi, 32)
+            .parallel(y)
+            .vectorize(x, 2);
```

Comment:
Oops, right. It's faster for vectorize by 4:
ARM:           0.290000ms(@415)
vs
ARM:           0.352000ms(@483).

Updated in 4a8f96.